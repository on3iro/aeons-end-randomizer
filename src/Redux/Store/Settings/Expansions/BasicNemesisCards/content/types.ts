import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.BasicNemesisCard }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardsContentStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        content: State
      }
    }
  }
}
