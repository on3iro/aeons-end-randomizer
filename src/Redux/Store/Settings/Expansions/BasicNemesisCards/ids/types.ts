import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardIdsStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        ids: string[]
      }
    }
  }
}
