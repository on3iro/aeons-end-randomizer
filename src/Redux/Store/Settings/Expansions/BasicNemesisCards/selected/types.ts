import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {
  TOGGLE_CARD = 'Settings/Expansions/BasicNemesisCards/TOGGLE_CARD',
  SET_TO_DB = 'Settings/Expansions/BasicNemesisCards/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/BasicNemesisCards/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/BasicNemesisCards/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/BasicNemesisCards/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/BasicNemesisCards/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/BasicNemesisCards/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedBasicNemesisCardsStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        selected: string[]
      }
    }
  }
}
