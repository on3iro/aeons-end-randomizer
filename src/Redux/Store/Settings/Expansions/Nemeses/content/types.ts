import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.Nemesis }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type NemesisContentStateSlice = {
  Settings: {
    Expansions: {
      Nemeses: {
        content: State
      }
    }
  }
}
