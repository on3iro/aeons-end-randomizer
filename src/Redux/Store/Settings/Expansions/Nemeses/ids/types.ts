import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type NemesisIdsStateSlice = {
  Settings: {
    Expansions: {
      Nemeses: {
        ids: string[]
      }
    }
  }
}
