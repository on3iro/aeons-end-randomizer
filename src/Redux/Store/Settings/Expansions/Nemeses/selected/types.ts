import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {
  TOGGLE = 'Settings/Expansions/Nemeses/TOGGLE',
  SET_TO_DB = 'Settings/Expansions/Nemeses/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/Nemeses/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/Nemeses/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/Nemeses/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/Nemeses/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/Nemeses/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedNemesesStateSlice = {
  Settings: {
    Expansions: {
      Nemeses: {
        selected: string[]
      }
    }
  }
}
