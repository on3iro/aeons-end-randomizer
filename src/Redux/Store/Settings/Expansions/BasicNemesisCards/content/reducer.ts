import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.basicNemesisCards,
  FR: AERData.normalizedData.FR.basicNemesisCards,
  PL: AERData.normalizedData.PL.basicNemesisCards,
  DE: AERData.normalizedData.DE.basicNemesisCards,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
