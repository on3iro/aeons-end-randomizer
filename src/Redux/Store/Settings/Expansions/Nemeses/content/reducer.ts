import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.nemeses,
  FR: AERData.normalizedData.FR.nemeses,
  PL: AERData.normalizedData.PL.nemeses,
  DE: AERData.normalizedData.DE.nemeses,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
