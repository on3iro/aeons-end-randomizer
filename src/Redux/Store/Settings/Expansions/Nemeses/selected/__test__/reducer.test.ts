import { Cmd, getCmd, getModel } from 'redux-loop'

import { State } from '../types'
import { actions } from '../actions'
import {
  getSelectedNemesesFromDB,
  setSelectedNemesesToDB,
} from '../sideEffects'

import { initialState, Reducer } from '../reducer'

const mockSelectedNemesesState: State = initialState

describe('Settings | Expansions | Nemeses | selected | reducer', () => {
  it('should return the initial state', () => {
    // @ts-ignore
    const result = Reducer(undefined, {})

    expect(result).toEqual(initialState)
  })

  it('should handle TOGGLE', () => {
    const selectedNemesesToSave = initialState.filter(
      (treasure) => treasure !== 'Maggoth'
    )

    const result = Reducer(mockSelectedNemesesState, actions.toggle('Maggoth'))

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toMatchSnapshot()
    expect(cmd).toEqual(
      Cmd.run(setSelectedNemesesToDB, {
        args: [selectedNemesesToSave],
        successActionCreator: actions.setToDBSuccessful,
        failActionCreator: actions.setToDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB', () => {
    const result = Reducer(initialState, actions.fetchFromDB())

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toEqual(initialState)

    expect(cmd).toEqual(
      Cmd.run(getSelectedNemesesFromDB, {
        successActionCreator: actions.fetchFromDBSuccessful,
        failActionCreator: actions.fetchFromDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB_SUCCESS for defined state', () => {
    const result = Reducer(
      mockSelectedNemesesState,
      actions.fetchFromDBSuccessful(['Maggoth'])
    )

    expect(getModel(result)).toEqual(['Maggoth'])
  })

  it('should handle FETCH_FROM_DB_SUCCESS for undefined state', () => {
    const result = Reducer(
      initialState,
      // @ts-ignore
      actions.fetchFromDBSuccessful(undefined)
    )

    expect(getModel(result)).toEqual(initialState)
  })

  it.todo('should handle FETCH_FROM_DB_FAILURE')
  it.todo('should handle SET_TO_DB')
  it.todo('should handle SET_TO_DB_SUCCESS')
  it.todo('should handle SET_TO_DB_FAILURE')
})
