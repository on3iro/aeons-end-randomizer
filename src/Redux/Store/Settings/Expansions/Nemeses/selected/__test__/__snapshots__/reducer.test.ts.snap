// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings | Expansions | Nemeses | selected | reducer should handle TOGGLE 1`] = `
Array [
  "CarapaceQueen",
  "CrookedMask",
  "PrinceOfGluttons",
  "Rageborne",
  "HordeCrone",
  "CloudedMesmer",
  "Necroswarm",
  "Bladius",
  "Deathmind",
  "FungalMesh",
  "MaelstromRisen",
  "SpawningHorror",
  "XaxosAscended",
  "Arachnos",
  "AgelessWalker",
  "Fenrix",
  "BlightLord",
  "WaywardOne",
  "ThriceDeadProphet",
  "Wraithmonger",
  "Stonemonger",
  "InfestedMutation",
  "ErythizaConsumed",
  "HeraldOfTheEnd",
  "TheChangelingNest",
  "CorruptionsCore",
  "TheWailing",
  "TheWanderer",
  "KnightOfShackles",
  "MaidenOfThorns",
  "GateWitch",
  "HollowCrown",
  "MagusOfCloaks",
  "UmbraTitan",
  "Experiment153",
  "ThiefOfDreams",
  "RisenThrall",
  "FountainOfSouls",
  "TheEndlessDecay",
  "Burrower",
  "Fortress",
  "TheBurningKor",
  "BishopOfScrolls",
  "BishopReturned",
  "FateShepherd",
  "Griefweaver",
  "MistRevealed",
  "MotherOfRust",
  "ParadoxOfBone",
  "ParadoxOfMyth",
  "ParadoxOfMythAndBone",
  "RustAmalgam",
  "ParadoxOfBone",
  "RustSuperion",
  "HazeFiend",
]
`;
